Zend/zend_builtin_functions.stub.php

```diff
@@ -6,23 +6,24 @@ function zend_version(): string {}
 
 function func_num_args(): int {}
 
-function func_get_arg(int $arg_num): mixed {}
+function func_get_arg(int $argument_number): mixed {}
```

Comment:
So ... as we're getting into renames now, I guess the bikeshedding is open...

I liked the `$arg_num` here more, it seems to fit better with the existing names (like func_num_args), which use these abbreviations.

Similarly, below we have various `str*` functions changed to use `$string` arguments -- while PHP is not very consistent in it's choice of `strpos` vs `str_replace`, it does seem to be consistent in not using `string_contains`.

I understand the desire to avoid abbreviations, but I think we also need to consider existing naming conventions.

Reply:
For me, it seems to be more important to bring consistency among all the parameter names than between function and parameter names. One of the reasons is that we can't change function names anymore (should it be possible, I believe we wouldn't use as much of abbreviations as before), but we have a very last opportunity to change parameter names in the right way.

If we stay with the example of `$str` vs `$string` parameter names, they are nearly equally used in the stubs all over the extensions (the current documentation would possibly show a bit different picture though). That's why I think it would be nice to stay with one format, since param names will now be present in executable code. Of course, `$needle` and `$haystack` has to stay. :D 

What do you think?

Reply:
For reference, here's how often parameter names are used, based on stubs: https://gist.github.com/nikic/b1223525b8bb0a1bef0f6b8a267318e9

My personal preference is to make names as short as sensible (but no shorter!) which includes the use of well-established abbreviations.

Just my 2c though. Maybe some other people like @cmb69 or @Girgias would like to comment.

Reply:
I have no strong preference either way, but having overly long parameter names might not be the best idea wrt. named parameters.

Reply:
I tried to look up [PEP 8](https://www.python.org/dev/peps/pep-0008/#prescriptive-naming-conventions) for Python but there is no mention of abbreviations. However one could interpret that the following two verses of the [Zen of Python](https://www.python.org/dev/peps/pep-0020/#id2):
>Explicit is better than implicit.
>Simple is better than complex.
To discourage the use of abbreviations.

Python do use some abbreviations but I think that's mostly due to the same issue the PHP can't rename some unfortunate function names.

Looking at the usage from stubs it seems to me that we mostly use the non abbreviated way for ``string``, ``length``, ``array`` and ``message`` which seems to cover most common argument names.

But I do agree that we maybe should consider using abbreviations if the argument currently has a long name (over 20 char?), or just renaming it to something better.

Reply:
@nikic @Girgias Thank you very much for the research :) I like the ideas presented by George, so using full names by default, but falling back to well-estabilished abbreviations if a name would be significantly longer otherwise. That said, I'm ok with using `arg/param` instead of `argument/parameter` if this is what you'd prefer (we can add more names to this list later), but I'd love if we could have `string`, `array`, and `message` in the same time. Would that work for you?

Additionally, I'd like to double-check if you are ok with making the procedural API use snake_case, and the OO methods use camelCase?

Reply:
I've just noticed that the list shows that the `*number` parameter name is currently used way more times than `*num`. So for me, it would be a bit weird to use `num` only in case of `func_get_arg()`. The `arg_number` parameter name would also be slightly weird because of mixing an abbreviated and non-abbreviated form (or is it just me?).

Either way, I'm not attached to the `$argument_number` name too much, so we can have `$arg_num` or `$arg_number`.

Reply:
What about using `$position`? It seems to me that this is the most concise and accurate name. `$arg_num` (and its variations) could mean "number of arguments". We could also go with `$arg_position`, but I believe `$position` is clear on its own. WDYT? I'll update the PR in the meanwhile :)

Reply:
I quite like ``$position`` :)

Reply:
Okay, let's go with this for now. We'll see how it goes...

> Additionally, I'd like to double-check if you are ok with making the procedural API use snake_case, and the OO methods use camelCase?

I'd probably go with using the same style as the function name uses. We have some OO APIs that use snake_case, in which case I'd also use snake_case for params. But most OO APIs do use camelCase.

Reply:
> I'd probably go with using the same style as the function name uses. We have some OO APIs that use snake_case, in which case I'd also use snake_case for params. But most OO APIs do use camelCase.

👍